Scene plotter visualiztion features

KieranRatcliffeInvertedAI opened the pull request:
Added the capability to change the colours of all agents, including the colour of their edges, to ScenePlotter. This should not force any existing implementations of ScenePlotter but give some visualization flexibility internally or externally if the user chooses to do so. 
Are there any other features that might be interesting to add to this PR such as adding agent path visualization to ScenePlotter?

KieranRatcliffeInvertedAI commented on invertedai/utils.py:
```diff
@@ -713,7 +718,17 @@ def plot_frame(self, idx, ax=None, numbers=False, direction_vec=False,
 
     def animate_scene(self, output_name=None, start_idx=0, end_idx=-1, ax=None,
```
> 

As I am going through the attributes and methods making docstrings, I have two questions as of now:
1. What does open_drive do? It appears to be a flag that enables different behaviour but what is the high level function of these different execution paths? 
2. It seems like there are a lot of parameters from the LocationResponse dataclass are input separately into the class. My question is why is this done separately instead of just inputting the LocationResponse directly?  Is there any use case where a user needs to modify these parameters after getting the LocationResponse individually since it is opening this tool up to misuse and seems overall clunky.

bzwartsenberg replied:
1. I don't know what `open_drive` does unfortunately. Maybe that's something Alireza added?

2. If you can refactor that I would say go for it, seems like a sensible thing to do!

KieranRatcliffeInvertedAI replied:
This was said in-person but for documentation purposes, I will be taking this feedback and doing a larger refactor of ScenePlotter.

adscib commented:
Do you have more changes coming? If so, I'll hold off on reviewing for now.

KieranRatcliffeInvertedAI commented:
> Do you have more changes coming? If so, I'll hold off on reviewing for now.

I do have some more changes coming and am doing a larger refactor.

adscib commented:
OK, let me know when it's ready.

KieranRatcliffeInvertedAI commented:
This PR is ready for a re-review. Noteable changes include:
1. All the open_drive relevant parameters as kwargs since this functionality will be used rarely.
2. Condensed the input parameters to a LocationResponse instead of inputting the parameters individually.
3. Added docustrings to the client-facing member functions of ScenePlotter.
4. Added data validation to the client-facing functions.
5. Added the ability to manually control the colours of the faces and edges of all agents.
6. Added sphinx documentation so ScenePlotter will be documented on our website.

bzwartsenberg reviewed (commented):
This looks awesome. I have not had time to look at the notebooks, but the updates to `ScenePLotter` look awesome, it's great to have some actual docstrings now. 

I could come back to the notebooks, if you want, but that seems like it might benefit from Frank / Robert's opinion more than mine.

KieranRatcliffeInvertedAI commented:
@bzwartsenberg coming back to this and seeing if there is time to review this?

bzwartsenberg reviewed (approved):
Again I have not looked carefully at the jupyter notebooks, but if the updates are only to propagate changes to the ScenePlotter itself that should be all good.

KieranRatcliffeInvertedAI commented:
@adscib I am going through old PR's and I've changed this to be more backwards compatible, requiring less concern about merging. Essentially this adds documentation and cleaned up the scene plotter, as well as adding left-hand option for visualizing Carla maps more easily. 
Let me know if you think there are any issues and if we can merge this into develop.

KieranRatcliffeInvertedAI commented:
@adscib Pinging this PR again to get the go-ahead to merge this into develop. Added some changes to update this branch to fit with changes to the develop branch.
